### DISCUSSION TITLE
钉钉工作通知启用后，redis里已经获取到了数据，但用户表里 ding_user_id 没有更新

### DISCUSSION THREAD

**Discussion by Spinestars** [hhyo/Archery#2069] [Q&A]
archery 系统版本是1.9.1

我在 common/utils/ding_api.py 中 get_ding_user_id() 看到了写入的过程，但除了 common/auth.py 中 authenticate_entry 看到调用了 get_ding_user_id ，没有在其它位置看到调用。

我的操作过程如下：
新创建了一个用户，它的 archery_username 就是钉钉的 userid，此时我没有手动写入 dingding_userid。
然后，我同步了钉钉用户到 redis里。redis 里缓存的kv是 archery_username: dingding_userid。
然后我再去看用户信息，发现这个用户的 dingding_userid 依然是空.

我该如何做，有人可以帮忙看下不？

**Comment by nick2wang** [comment]
新建的用户登录成功后才会同步写入dingding_user_id

**Reply by Spinestars** [reply]
嗯，我确实做了这个尝试（问题里忘了说这一点了），但是不知道为啥依然没有写入，而且也没有报错。。。

**Reply by Spinestars** [reply]
我看到调用 get_ding_user_id 是在  ! SysConfig().get("enforce_2fa") 中，但我开启了 SysConfig().get("enforce_2fa")。这会不会是问题点？

**Reply by Spinestars** [reply]
![image](https://user-images.githubusercontent.com/7272900/222111352-a4eda608-7a5e-48ba-83e2-e4cadbdc1a92.png)
![image](https://user-images.githubusercontent.com/7272900/222111428-a1cee0de-b944-4e08-93fb-c3b76a55a9a8.png)

**Reply by nick2wang** [reply]
开启了2fa的话用户登录就是走api，api登录流程应该漏了get_ding_user_id，你可以先加上试试
https://github.com/hhyo/Archery/blob/b18de11fb3d5295f92c32eedd5f4b2c2e78fb72a/sql_api/api_user.py#L419-L422

**Reply by Spinestars** [reply]
OK，我试下
